PCSX2/pcsx2
# Optionally use system libs

### Description of Changes
Adds the ability to use system libs for a number of dependencies

You can now specify `-DUSE_SYSTEM_LIBS=ON` to force PCSX2 to require system libs and fail if they can't be found.  You can also specify `-DUSE_SYSTEM_LIBS=OFF` to not attempt to use system libs at all.  The default (`AUTO`) uses system libs if they can be found, and bundled libs otherwise.  In addition, there are separate overrides for each of the optionally-bundled libraries.

### Rationale behind Changes
Users may prefer system libs for a number of reasons, so it's nice to have the option.

### Suggested Testing Steps
- Make sure save states are loadable with system libzip (tested by me, but might as well check)
- Make sure system SDL2 works in Qt (not tested by me)
- Make sure cmake rejects outdated packages (SDL2 is probably easiest to test, plenty of linux distros have a < 2.0.22 copy)
  - In this case, make sure the build continues with the bundled SDL in the default configuration, and fails if you specify `-DUSE_SYSTEM_SDL2=ON`
